### PR TITLE
fix: external links in mascot announcements now work

### PR DESCRIPTION
**Pull request by Doug-Reed**
fixes bug where add to home logic short-circuited external links

----

PR considerations checklist:

- [X] Updates `CHANGELOG.md` to reflect this PR's change.
- [X] This Contribution is under the terms of Individual [Contributor License Agreements][] (and also Corporate Contributor License Agreements to the extent applicable) appearing in the [Apereo CLA roster][].

[Apereo CLA roster]: http://licensing.apereo.org/completed-clas
[Contributor License Agreements]: https://www.apereo.org/licensing/agreements

### CODE REVIEW

File: components/portal/messages/controllers.js [hunk 1]
```diff
@@ -688,6 +688,8 @@ define(['angular'], function(angular) {
             $rootScope.addPortletToHome(fName);
             actionButton.label = 'On your home';
             actionButton.disabled = true;
+          } else {
+            $window.open(actionButton.url, '_blank');
```
Comment (ChristianMurphy) [inline components/portal/messages/controllers.js]: does `window.open` automatically work like a link with `rel="noopener noreferrer"`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -33,6 +33,7 @@ a dismissible-for-session toast message prompting users to add the app to their
 home page, instead of a button. (#684)
 + `vm.showMessagesFeatures` variable moved out of default and into more 
 logically appropriate spot (#694)
++ External links in mascot announcements now work again. (#697)
```
Comment (apetro) [inline CHANGELOG.md]: I'd categorize this as "Fixed".

File: components/portal/messages/controllers.js (hunk 1 above)
Comment (apetro) [reply components/portal/messages/controllers.js]: Will the URL always be external-to-the-portal, or might it be sensible to mascot-announce a path within the portal that's not a widget to add to home?

File: components/portal/messages/controllers.js [hunk 3]
```diff
@@ -438,11 +438,11 @@ define(['angular'], function(angular) {
 
     .controller('AnnouncementsController', ['$q', '$log', '$filter',
       '$sessionStorage', '$scope', '$rootScope', '$document', '$sanitize',
-      '$mdDialog', 'miscService',
-      'messagesService', 'PortalAddToHomeService', 'MISC_URLS',
+      '$mdDialog', 'miscService', 'messagesService',
```
Comment (ChristianMurphy) [inline components/portal/messages/controllers.js]: :art: `441:53  error  Trailing spaces not allowed  no-trailing-spaces`

https://travis-ci.org/uPortal-Project/uportal-app-framework/jobs/340695770#L720

Comment (Doug-Reed) [reply components/portal/messages/controllers.js]: Fixed.

File: components/portal/messages/controllers.js (hunk 1 above)
Comment (Doug-Reed) [reply components/portal/messages/controllers.js]: I explicitly added the re=noopener noreferrer and logic to distinguish between internal and external links.

**Pull request merged**